### PR TITLE
Add jobs: ChannelTopUp, UncooperativeCloseRequest.

### PR DESCRIPTION
**Pull request by dzeckelev**
- ClientPreChannelTopUp
- ClientAfterChannelTopUp
- ClientPreUncooperativeCloseRequest
- ClientAfterUncooperativeCloseRequest
Resolves BV-362.

### CODE REVIEW

File: proc/worker/client.go [hunk 1]
```diff
@@ -37,6 +38,27 @@ func (w *Worker) clientPreChannelCreateCheckDeposit(
 	return deposit, nil
 }
 
+func (w *Worker) checkReceiptBalance(ch *data.Channel) error {
+	// check channel status
+	switch ch.ChannelStatus {
+	case data.ChannelActive, data.ChannelPending:
+	default:
+		return ErrInvalidChStatus
```
Comment (furkhat) [inline proc/worker/client.go]: just if
```go
if ch.ChannelStatus != data.ChannelActive && ch.ChannelStatus != data.ChannelPending {
        return ErrInvalidChStatus
```

File: proc/worker/client.go [hunk 2]
```diff
@@ -37,6 +38,27 @@ func (w *Worker) clientPreChannelCreateCheckDeposit(
 	return deposit, nil
 }
 
+func (w *Worker) checkReceiptBalance(ch *data.Channel) error {
```
Comment (furkhat) [inline proc/worker/client.go]: function name does not correspond to its body (it only corresponds to the last `if` statement). Ambiguous for the caller, please change the name.

File: proc/worker/client.go [hunk 3]
```diff
@@ -290,3 +313,211 @@ func (w *Worker) ClientAfterEndpointMsgSOMCGet(job *data.Job) error {
...
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	opts := bind.NewKeyedTransactor(key)
+	opts.Context = ctx
+	if gasPrice != 0 {
+		opts.GasPrice = new(big.Int).SetUint64(gasPrice)
+	}
+
+	// TODO(maxim) set gas limit from conf
```
Comment (furkhat) [inline proc/worker/client.go]: config is merged, you can already do it

File: proc/worker/client.go [hunk 4]
```diff
@@ -290,3 +313,211 @@ func (w *Worker) ClientAfterEndpointMsgSOMCGet(job *data.Job) error {
...
+// ClientPreChannelTopUp checks client balance and creates transaction
+// for increase the channel deposit.
+func (w *Worker) ClientPreChannelTopUp(job *data.Job) error {
+	var jdata ClientPreChannelTopUpData
+	if err := parseJobData(job, &jdata); err != nil {
+		return err
+	}
+
+	var ch data.Channel
+	if err := data.FindByPrimaryKeyTo(w.db.Querier, &ch,
```
Comment (furkhat) [inline proc/worker/client.go]: move to `selectors.go`

File: proc/worker/client.go [hunk 5]
```diff
@@ -290,3 +313,211 @@ func (w *Worker) ClientAfterEndpointMsgSOMCGet(job *data.Job) error {
...
+		jdata.Channel); err != nil {
+		return err
+	}
+
+	var offer data.Offering
+	if err := data.FindByPrimaryKeyTo(w.db.Querier, &offer,
```
Comment (furkhat) [inline proc/worker/client.go]: use `w.offering(pk)` defined in `selectors.go`

File: proc/worker/client.go [hunk 6]
```diff
@@ -290,3 +313,211 @@ func (w *Worker) ClientAfterEndpointMsgSOMCGet(job *data.Job) error {
...
+		ch.Offering); err != nil {
+		return err
+	}
+
+	var acc data.Account
+	if err := data.FindOneTo(w.db.Querier, &acc,
```
Comment (furkhat) [inline proc/worker/client.go]: use `w.account` defined in selectors.go

File: proc/worker/client.go [hunk 7]
```diff
@@ -290,3 +313,211 @@ func (w *Worker) ClientAfterEndpointMsgSOMCGet(job *data.Job) error {
...
+		return err
+	}
+
+	return w.clientPreChannelTopUpSaveTx(job, &ch, &acc, &offer,
+		jdata.GasPrice, new(big.Int).SetUint64(deposit))
+}
+
+// ClientAfterChannelTopUp updates deposit of a channel.
+func (w *Worker) ClientAfterChannelTopUp(job *data.Job) error {
+	return w.afterChannelTopUp(job, false)
```
Comment (furkhat) [inline proc/worker/client.go]: redundant complexity with flag, just send job type:
```go
return w.afterChannelTopUp(job, data.JobClientAfterChannelTopUp)
```

File: proc/worker/client.go [hunk 8]
```diff
@@ -290,3 +313,211 @@ func (w *Worker) ClientAfterEndpointMsgSOMCGet(job *data.Job) error {
...
+	}
+
+	var ch data.Channel
+	if err := data.FindByPrimaryKeyTo(w.db.Querier, &ch,
+		jdata.Channel); err != nil {
+		return err
+	}
+
+	var offer data.Offering
+	if err := data.FindByPrimaryKeyTo(w.db.Querier, &offer,
```
Comment (furkhat) [inline proc/worker/client.go]: use selector functions

File: proc/worker/client.go (hunk 1 above)
Comment (dzeckelev) [reply proc/worker/client.go]: switch looks simpler and better perceived

Comment (furkhat) [reply proc/worker/client.go]: no, it does not :)

File: proc/worker/errors.go [hunk 9]
```diff
@@ -6,5 +6,10 @@ import "errors"
 var (
 	ErrInvalidJob       = errors.New("unexpected job type or job related type")
 	ErrNotEnoughBalance = errors.New("not enough PSC balance for offering")
-	ErrNoSupply         = errors.New("no supply")
+	ErrChReceiptBalance = errors.New("receipt balance is more than a deposit")
```
Comment (ababo) [inline proc/worker/errors.go]: Mb "is greater"?

**Pull request merged**